default-constructed-buffers/default-constructed-buffers.md

```diff
@@ -53,18 +54,21 @@ class buffer {
 The template arguments should remain the same, so that the argument can be
 rebound to a new `buffer` instance later using the copy constructor.
 
-The `is_valid()` call would allow the programmer to query whether or not
-the buffer can be used, i.e. whether or not it was constructed with a range of
-size zero. The explicit conversion operator would call this same function but
-allow its use in `if` statements.
+The `has_storage()` call would allow the programmer to query whether or not
+the buffer can be used, i.e. whether or not it was default-constructed. The
+explicit conversion operator would call this same function but allow its use
+in `if` statements.
 
 Requesting access from a default-constructed buffer should throw an exception.
-It is not meaningful to use a zero-sized allocation on-device. Since there is
-no allocation associated with the `buffer`, `cl::sycl::buffer::set_final_data`
-and `cl::sycl::buffer::set_write_back` should behave as if the `buffer` had a
-final pointer of `nullptr` at all times. The other functions in the `buffer`
-API should behave as though the buffer were constructed with a `range` of size
-zero and otherwise behave normally.
+It is not meaningful to use a default-constructed buffer on device. Since there
+is no allocation associated with a default-constructed `buffer`,
+`cl::sycl::buffer::set_final_data` and `cl::sycl::buffer::set_write_back`
+should behave as if the `buffer` had a final pointer of `nullptr` at all times.
+The other functions in the `buffer` API should behave as though the buffer were
+constructed with a `range` of size zero and otherwise behave normally.
+
+Explicitly constructing a buffer with a range of zero is not allowed. This does
+not correspond to a valid device allocation.
```

Comment:
We should extend this statement to also mention host allocations, where applicable.

Reply:
I guess here I was contrasting the implementation-defined nature of `malloc(0)` compared to the totally-forbidden error from `clCreateBuffer(context, CL_MEM_READ_WRITE, 0ull, nullptr, &err)`. Would changing it to say "This does not correspond to a valid allocation" be an acceptable fix, since it's just flat-out not allowed?

Reply:
Yeah, this sounds good to me 👍